src/utils/settings.cpp

```diff
@@ -32,31 +34,56 @@ namespace mp = multipass;
 namespace
 {
 const auto file_extension = QStringLiteral("conf");
+const auto daemon_root = QStringLiteral("local");
 const auto petenv_name = QStringLiteral("primary");
+
 std::map<QString, QString> make_defaults()
+{ // clang-format off
+    return {{mp::petenv_key, petenv_name},
+            {mp::driver_key, mp::platform::default_driver()}};
+} // clang-format on
+
+/*
+ * We make up our own file names to:
+ *   a) avoid unknown org/domain in path;
+ *   b) write daemon config to a central location (rather than user-dependent)
+ * Examples:
+ *   - ${HOME}/.config/multipass/multipass.conf
+ *   - /root/.config/multipass/multipassd.conf
+ */
+QString file_for(const QString& key) // the key should have passed checks at this point
 {
-    return {{mp::petenv_key, petenv_name}};
+    static const auto file_path_base = QStringLiteral("%1/%2.%3"); // static consts ensure these stay fixed
+    static const auto client_dir_path = QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation);
+    static const auto daemon_dir_path = QStringLiteral("/etc/xdg"); // temporary
+    static const auto client_file_path = file_path_base.arg(client_dir_path, mp::client_name, file_extension);
+    static const auto daemon_file_path = file_path_base.arg(daemon_dir_path, mp::daemon_name, file_extension);
+
+    assert(key.startsWith(daemon_root) || key.startsWith("client"));
+    return key.startsWith(daemon_root) ? daemon_file_path : client_file_path;
 }
 
-std::unique_ptr<QSettings> persistent_settings()
+std::unique_ptr<QSettings> persistent_settings(const QString& key)
 {
-    static const auto file_path = QStringLiteral("%1/%2.%3") // make up our own file name to avoid org/domain in path
-                                      .arg(QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation)) // dir
-                                      .arg(QCoreApplication::applicationName())
-                                      .arg(file_extension);
-    static const auto format = QSettings::defaultFormat(); // static consts to make sure these stay fixed
-
-    return std::make_unique<QSettings>(file_path, format); /* unique_ptr to circumvent absent copy-ctor and no RVO
-                                                              guarantee (until C++17) */
+    static const auto format = QSettings::defaultFormat(); // static const to make sure these stay fixed
+
+    return std::make_unique<QSettings>(file_for(key), format); /* unique_ptr to circumvent absent copy-ctor and no RVO
+                                                                  guarantee (until C++17) */
+}
+
+bool is_unreadable(const QString& filename)
```

Comment:
I'd probably make this `is_readable()` and move the negation to the consumer.

Reply:
Yeah I understand your point. `is_readable` would be more elementary and so more general. But here `is_unreadable` means the file *exists and cannot be read*. I think this makes sense in English, as you would not say a non-existing file is unreadable. But, if I negated that, *I'd be saying non-existing files were readable*. While that can be argued to be true (you know exactly what the contents of an absent file are, and you can "read" it's absence provided you have read permission to the parent), it sounds less intuitive to me. But ok, if you insist I don't mind doing that. Give me a verdict.

Reply:
So `is_unreadable("non-existent-file");` is false… i.e. a missing file is "not unreadable"? I'm getting confused here ;)

If this is just about permissions, then I'd say we need a more descriptive name for the function, then, or even move the `.exists()` out of it.

Reply:
Yeah I thought it made sense to say that unreadable required reading to be locked, which required the file to be there. But ok, I'll rework this then.